version: "3.3"

services:

  traefik:
    image: "traefik:latest"
    container_name: "traefik"
    labels:
      - "traefik.http.middlewares.https-redirect.redirectscheme.scheme=https"
      - "traefik.http.middlewares.tlsredirect.redirectscheme.scheme=https"
      - "traefik.http.middlewares.auth.basicauth.usersfile=/auth/management.auth"
      - "traefik.http.middlewares.compress.compress=true"
      - "traefik.http.middlewares.errorhandler.errors.status=400-599"
      - "traefik.http.middlewares.errorhandler.errors.service=resolver-global-errorpages"
      - "traefik.http.middlewares.errorhandler.errors.query=/{status}.html"
      - "traefik.http.middlewares.inflight.inflightreq.amount=10"
      - "traefik.http.middlewares.retry.retry.attempts=4"
      # HTTP
      - "traefik.http.routers.traefik-http.rule=Host(`traefik.odex.be`) || Host(`traefik.pbfp.xyz`)"
      - "traefik.http.routers.traefik-http.entrypoints=http"
      - "traefik.http.routers.traefik-http.middlewares=tlsredirect"
      # HTTPS
      - "traefik.http.routers.traefik-https.rule=Host(`traefik.odex.be`) || Host(`traefik.pbfp.xyz`)"
      - "traefik.http.routers.traefik-https.entrypoints=https"
      - "traefik.http.routers.traefik-https.tls=true"
      - "traefik.http.routers.traefik-https.tls.certResolver=letsencrypt"
      - "traefik.http.routers.traefik-https.service=api@internal"
      - "traefik.http.routers.traefik-https.middlewares=auth"
    ports:
      - "80:80"
      - "443:443"
      - "8080:8080"
    volumes:
      - "./letsencrypt:/letsencrypt"
      - "/var/run/docker.sock:/var/run/docker.sock:ro"
      - "./logs/:/logs"
      - "./traefik.yaml:/etc/traefik/traefik.yaml"
    networks:
      - web

networks:
  web:
   external: true
